packit/command_handler.py

```diff
@@ -157,3 +159,7 @@ def run_command(
         if return_output:
             return out
         return None
+
+    def clean(self):
+        logger.info("Deleting sandcastle pod.")
+        self.sandcastle.delete_pod()
```

Comment:
This is good but there is very little guarantee that it's going to be called in case an exception occurs somewhere. Could this be improved by deleting the pod also in `sandcastle.__del__()`, which has guarantees to be called whenever a Sandcastle object is deleted?

Reply:
my experience with destructors in python was often cumbersome, let's give this a try

Reply:
btw, the pod has a timeout of 10 minutes (`sleep 600`), so even if it fails to be cleaned up this way, it should disappear sooner or later

(on the other hand, PVs would keep lingering)